### PR TITLE
PowerShell deployment improvements

### PR DESCRIPTION
**Pull request by jamescarpinter**
Bug fixes and improvements to streamline deployment via PowerShell:
* Add gitattributes file (based on DNX source) and include specific rule for *.sh files (need to use Unix-style line ending otherwise they don't run as expected).
* Switch to Oracle Java 8 as the Azure Storage client library throws exceptions when running on the OpenJDK.
* Remove the front-end from the deployment script (not up to date with the new MRP PartsUnlimited scenario).
* Add retry loop to MongoDB provisioning script. When run as part of a Azure PowerShell VM deployment script, sometimes it takes a bit of time after install before MongoDB starts responding to requests.
* Update PowerShell script to upload dependencies and reuse subscriptions / storage accounts.

### CODE REVIEW

File: deploy/Install-MRP-app.sh [hunk 1]
```diff
@@ -1,45 +1,28 @@
...
-sed -i s/8080/9080/g /etc/tomcat7/server.xml
-
-# Download the client WAR file
-wget ${AzureResource}mrp.war
-
-# Wait for 10 seconds to make sure previous step is completed
-sleep 10
-
-# Copy WAR file to Tomcat directory for auto-deployment
-cp mrp.war /var/lib/tomcat7/webapps
```
Comment (dtzar) [inline deploy/Install-MRP-app.sh]: I know we don't use the front-end web app for the integration service, but we do use it for other HOLs, so I'd like to keep it.  If this introduces problems for the integration lab, I'd like to know what they are.

File: deploy/Install-MRP-app.sh [hunk 2]
```diff
@@ -1,30 +1,31 @@
-﻿#!/bin/bash
+#!/bin/bash
+
+# Auto-accept license for Java
+echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | /usr/bin/debconf-set-selections
```
Comment (OguzPastirmaci) [inline deploy/Install-MRP-app.sh]: Missing sudo after pipe:
echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | sudo /usr/bin/debconf-set-selections

File: deploy/Install-MRP-app.sh [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
 
 # Install packages
+add-apt-repository ppa:webupd8team/java
```
Comment (OguzPastirmaci) [inline deploy/Install-MRP-app.sh]: Can you add -y in the end so it won't ask for confirmation:
add-apt-repository ppa:webupd8team/java -y

File: deploy/Install-MRP-app.sh [hunk 4]
```diff
@@ -1,30 +1,31 @@
...
-apt-get install openjdk-8-jre -y
+apt-get install oracle-java8-installer -y
 apt-get install mongodb -y
 apt-get install tomcat7 -y
 apt-get install wget -y
 
-# Set Java environment variables
-export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
-export PATH=$PATH:/usr/lib/jvm/java-8-openjdk-amd64/bin
+export JAVA_HOME=/usr/lib/jvm/java-8-oracle
```
Comment (OguzPastirmaci) [inline deploy/Install-MRP-app.sh]: Tomcat is sometimes picky about finding the right path. Creating a symlink helps this:
ln -s /usr/lib/jvm/java-8-oracle /usr/lib/jvm/default-java

Comment (jamescarpinter) [inline deploy/Install-MRP-app.sh]: Fixed - added in symlink as suggested.

File: deploy/Install-MRP-app.sh (hunk 3 above)
Comment (jamescarpinter) [inline deploy/Install-MRP-app.sh]: Fixed - added in command line switch as suggested.

File: deploy/Install-MRP-app.sh (hunk 2 above)
Comment (jamescarpinter) [inline deploy/Install-MRP-app.sh]: Fixed - added in sudo as suggested.

File: deploy/Install-MRP-app.sh (hunk 1 above)
Comment (jamescarpinter) [inline deploy/Install-MRP-app.sh]: Fixed - reverted removal of Tomcat/MRP deployment as suggested.

**Pull request merged**